### DISCUSSION TITLE
NVIDIA GPU PCI pass through not working for Talos VM image

### DISCUSSION THREAD

**Discussion by purankhoeval** [siderolabs/talos#9558] [Q&A]
Hello! I'm using LXD to spin up a VM and have attached a GPU through device config and able to see the passthrough GPU attached through VFIO-PCI driver. ( I have blacklisted NVIDIA Host drivers) 

Further I have installed Talos OS image built with the requisite system extensions for Kata containers, NVIDIA container tool kit and open source GPU. The modules are patched with the patch file described in the Talos docs however in Talos console I see the error as NVIDIA kernel modules are not loaded and NVRM: This PCI I/O region assigned to your NVIDIA device is invalid.
 
Am I fundamentally missing something here. New in this space, any help is much appreciated!

**Comment by jlengelbrecht** [comment]
@purankhoeval were you able to get this figured out? Running into a similar issue trying to pass through a gpu to talos VM running proxmox

**Reply by purankhoeval** [reply]
No,I could not resolve the issue so moved on with LibVirt VM and RKE2 as I ultimately needed device support for vGPU instead of whole device passthrough.Here is the thread where you can get in touch with someone who got it working with Proxmox and PCI pass through https://www.reddit.com/r/Proxmox/s/gf9JhpEB2f

**Reply by jlengelbrecht** [reply]
This helps a lot! thank you. Actually doing this with VM's created by terraform in proxmox. This way once I get my cluster in a running state easily then I can add crossplane at future date to spin up additional cluster nodes as needed if my work loads grow past what I initially setup. Thanks again
